eth/main.cpp

```diff
@@ -1160,17 +1162,21 @@ int main(int argc, char** argv)
 
 	if (jsonRPCURL > -1 || ipc)
 	{
-		auto safeConnector = new SafeHttpServer(jsonRPCURL, "", "", SensibleHttpThreads);
-		safeConnector->setAllowedOrigin(rpcCorsDomain);
-		jsonrpcConnector.reset(safeConnector);
-		jsonrpcServer = make_shared<dev::WebThreeStubServer>(*jsonrpcConnector.get(), web3, make_shared<SimpleAccountHolder>([&](){ return web3.ethereum(); }, getAccountPassword, keyManager, authenticator), vector<KeyPair>(), keyManager, *gasPricer);
+		auto httpConnector = new SafeHttpServer(jsonRPCURL, "", "", SensibleHttpThreads);
+		auto ipcConnector = new IpcServer("geth");
+		auto webthreeFace = new dev::WebThreeStubServer(web3, make_shared<SimpleAccountHolder>([&](){ return web3.ethereum(); }, getAccountPassword, keyManager, authenticator), vector<KeyPair>(), keyManager, *gasPricer);
+		
+		jsonrpcServer.reset(new ModularServer<dev::WebThreeStubServer, AbstractDb>(webthreeFace, new LevelDBFace()));
+		httpConnector->setAllowedOrigin(rpcCorsDomain);
+		jsonrpcServer->addConnector(httpConnector);
+		jsonrpcServer->addConnector(ipcConnector);
 		if (jsonRPCURL > -1)
 			jsonrpcServer->StartListening();
-		jsonrpcServer->enableIpc(ipc);
+		ipcConnector->StartListening();
```

Comment:
Should do that only if `ipc == true`. I'd suggest not calling `addConnector` if corresponding ipc or http rpc is not enabled